packages/core/core-flows/src/product/workflows/update-products.ts

```diff
@@ -131,6 +141,25 @@ function findProductsWithSalesChannels({
   return !input.update?.sales_channels ? [] : productIds
 }
 
+function findProductsWithShippingProfile({
```

Comment:
If we require shipping profile on product creation all will have it, so we can save ourselves from this loop